bait-files/src/test/java/com/google/sps/UserTest.java

```diff
@@ -0,0 +1,131 @@
+      new LocalServiceTestHelper(new LocalUserServiceTestConfig())
+          .setEnvIsAdmin(true).setEnvIsLoggedIn(true).setEnvEmail("andrew").setEnvAuthDomain("gmail.com");;
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  // Database is empty, so username should not be taken
+  @Test
+  public void emptyUsernameTakenTest() throws IOException {
+        HttpServletRequest request = mock(HttpServletRequest.class);       
+        HttpServletResponse response = mock(HttpServletResponse.class);    
+
+        when(request.getParameter("username")).thenReturn("Andrew");
+
+        StringWriter stringWriter = new StringWriter();
+        PrintWriter writer = new PrintWriter(stringWriter);
+
+        when(response.getWriter()).thenReturn(writer);
+        new UsernameTakenServlet().doGet(request, response);
+
+        verify(request, atLeast(1)).getParameter("username"); // only if you want to verify username was called...
+        writer.flush(); // it may not have been flushed yet...
+        assertTrue(stringWriter.toString().contains("false"));
+  }
+
+  // Username field is null, so return that the username is taken
+  @Test
+  public void nullUsernameTest() throws IOException {
+        HttpServletRequest request = mock(HttpServletRequest.class);       
+        HttpServletResponse response = mock(HttpServletResponse.class);
+
+        when(request.getParameter("username")).thenReturn(null);
+
+        StringWriter stringWriter = new StringWriter();
+        PrintWriter writer = new PrintWriter(stringWriter);
+
+        when(response.getWriter()).thenReturn(writer);
+        new UsernameTakenServlet().doGet(request, response);
+
+        verify(request, atLeast(1)).getParameter("username"); // only if you want to verify username was called...
+        writer.flush(); // it may not have been flushed yet...
+        assertTrue(stringWriter.toString().contains("true"));
+  }
+
+
+  // Username field is an empty string, so return that the username is taken
+  @Test
+  public void emptyStringUsernameTest() throws IOException {
+        HttpServletRequest request = mock(HttpServletRequest.class);       
+        HttpServletResponse response = mock(HttpServletResponse.class);
+
+        when(request.getParameter("username")).thenReturn("");
+
+        StringWriter stringWriter = new StringWriter();
+        PrintWriter writer = new PrintWriter(stringWriter);
+
+        when(response.getWriter()).thenReturn(writer);
+        new UsernameTakenServlet().doGet(request, response);
+
+        verify(request, atLeast(1)).getParameter("username"); // only if you want to verify username was called...
+        writer.flush(); // it may not have been flushed yet...
+        assertTrue(stringWriter.toString().contains("true"));
+  }
+
+  // Add a user and ensure there is a new entry in database
+  @Test
+  public void addUser() throws IOException, ServletException {
```

Comment:
As mentioned way above, I'd move this into a new test file called LoginServletTest.java, and naming wise you could name this test something like:
doPost_addsUserToDatabase

I'd say it's fine to break the three part naming convention when you only need one test and therefore don't have a variety of setup conditions.  If you did want a three part name, you could always go with something like:
doPost_forAnyInput_addsUserToDatabase

Reply:
Renamed